### PR TITLE
[CLI] Cannot run make localnet_client_debug: Cannot initialise the keybase with the validator keys: Unable to find YAML file

### LINKED ISSUE #516: [CLI] Cannot run `make localnet_client_debug`: Cannot initialise the keybase with the validator keys: Unable to find YAML file
## Objective

After merging `main` into my #491 PR, I have noticed that the `CLI` stopped working.

We need to fix it (I actually already did it in my branch last night/this morning since this was blocking me)

## Origin Document

Run `make localnet_up` and then when the stack is up and running try `make localnet_client_debug`

You'll get this error: 
image

This is due to the fact that we cannot reference the `private-keys.yaml` file locally by path in K8S

## Goals

- Fix the error
- Make sure that it works both locally and in K8S

## Deliverable

- [ ] Updated Keybase/CLI code

## Non-goals / Non-deliverables

- Changes in logic

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)
- [ ] **K8S Localnet**

---

**Creator**: @deblasis 
**Co-Owners**:

### PR DESCRIPTION
**Pull request by deblasis**
## Description

This PR introduces the ability to fetch private keys when running in Kubernetes

## Issue

Fixes #516 

## Type of change

Please mark the relevant option(s):

- [x] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Added the ability to detect if running inside a Kubernetes Pod
- Introduced logical switch to handle parsing of the debug private keys from a local file OR from Kubernetes 
- Introduced the Kubernetes manifests necessary to grant permissions to the debug client (least privilege)

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: shared/k8s/debug.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func FetchValidatorPrivateKeys(clientset *kubernetes.Clientset) (map[string]string, error) {
+	var validatorKeysMap = make(map[string]string)
```
Comment (Olshansk) [inline shared/k8s/debug.go:11]: Haven't seen `var ` in a while.

Option 1: `validatorKeysMap := `
Option 2: named params

File: shared/k8s/debug.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	private_keys_secret, err := clientset.CoreV1().Secrets("default").Get(context.TODO(), "v1-localnet-validators-private-keys", metav1.GetOptions{})
```
Comment (Olshansk) [inline shared/k8s/debug.go:13]: We do 🐫 here: `privKeysSecret`

Comment (Olshansk) [inline shared/k8s/debug.go:13]: Can you move `v1-localnet-validators-private-keys` into a constant with a comment that it's the name of an expected resource

File: shared/k8s/debug.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	for id, privHexString := range private_keys_secret.Data {
+		validatorKeysMap[id] = string(privHexString)
```
Comment (Olshansk) [inline shared/k8s/debug.go:19]: Is the hex string stored as utf8 encoded bytes in `Data`? If so, just add a comment

File: build/localnet/manifests/service-accounts.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: debug-client-account
```
Comment (Olshansk) [inline build/localnet/manifests/service-accounts.yaml:4]: WHy not specify the namespace for the ServiceAccount too?

File: app/client/keybase/debug/keystore.go [hunk 5]
```diff
@@ -120,6 +108,58 @@ func InitialiseDebugKeybase() error {
...
+		ApiVersion string            `yaml:"apiVersion"`
+		Kind       string            `yaml:"kind"`
+		MetaData   map[string]string `yaml:"metadata"`
+		Type       string            `yaml:"type"`
+		StringData map[string]string `yaml:"stringData"`
+	}
+	if err := yaml.Unmarshal(yamlData, &config); err != nil {
+		return nil, err
+	}
+	var validatorKeysMap = make(map[string]string)
```
Comment (Olshansk) [inline app/client/keybase/debug/keystore.go:155]:
```suggestion
	validatorKeysMap := make(map[string]string)
```

**Review by Olshansk — CHANGES_REQUESTED** [review]
Just a few nites. Let's tend to it and merge it in.

Also, while looking into this, I realized that chatGPT can be great for helping write k8s manifest templates

File: shared/k8s/debug.go (hunk 3 above)
Comment (deblasis) [reply shared/k8s/debug.go:22]: The documentation says it's base64 encoded but it's not. simply casting to string returns the actual encoded private key. 

image

An experiment I just did:

image

pk represents the base64 decoded value from what is not base64 encoded

File: build/localnet/manifests/service-accounts.yaml (hunk 4 above)
Comment (deblasis) [reply build/localnet/manifests/service-accounts.yaml:4]: it's redundant but for consistency let's add it

File: shared/k8s/debug.go (hunk 3 above)
Comment (Olshansk) [reply shared/k8s/debug.go:23]: So we're encoding it to base64, but then casting it to a byte slice which is just "a weird thing to do" unless you want to get the byte count of the string or something.

Sound correct? Can you add a TODO/BUG next to the `Data` attribute so we don't lose this.

**Comment by deblasis** [comment]
> Just a few nites. Let's tend to it and merge it in.

@Olshansk it should be good to go

> Also, while looking into this, I realized that chatGPT can be great for helping write k8s manifest templates

I have snippets in VSCode from my Hashicorp days for templates, the tooling improved A LOT over the years.

**Review by Olshansk — APPROVED** [review]
Let's tend to https://github.com/pokt-network/pocket/pull/517/files#r1107913640 later

File: shared/k8s/debug.go (hunk 3 above)
Comment (deblasis) [reply shared/k8s/debug.go:23]: No that's K8S documentation, they return a []byte and say that it's base64 encoded but it's not.

**Pull request merged**